### PR TITLE
Memory: fix en pipeline for read ports

### PR DESCRIPTION
**Pull request by ekiwi**
The `enable` signal on a read port is never used by `treadle` and thus this bug was never discovered. However, I am using it for my symbolic execution extension.

This fixes the fact that instead of creating a enable and a address pipeline the old code would create two address pipelines.
During the course of fixing this I merge the functions for read and write ports into one as their only difference was the latency which can be a function parameter.

### CODE REVIEW

**Comment by ekiwi** [comment]
In case anyone is wondering: the changes from `readerName`/`writerName` to `name` allowed me to keep the 120 character horizontal limit. `name` should not be ambiguous and in the case of a read/write port, it is actually more correct.

File: src/main/scala/treadle/executable/Memory.scala [hunk 1]
```diff
@@ -417,33 +417,25 @@ object Memory {
       }
     }
 
-    /*
+    /**
       * Makes a read chain of pipeline registers.  These must be ordered reg0/in, reg0, reg1/in ... regN/in, regN
```
Comment (chick) [inline src/main/scala/treadle/executable/Memory.scala]: Maybe this comment should be edited to just saw a chain of pipeline registers

Comment (ekiwi) [reply src/main/scala/treadle/executable/Memory.scala]: What exactly would you like the comment to say?

Comment (chick) [reply src/main/scala/treadle/executable/Memory.scala]: "Makes a chain of ..."

**Pull request merged**